File: 0229 majority-element-ii/solution.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        List<Integer> l = new ArrayList<>();
        int count1=0,count2=0;
        int e1=0,e2=0;
        for(int i=0;i<nums.length;i++){
            int num=nums[i];
            if(e1==num) count1++;
            else if(e2==num ) count2++;
            else if(count1==0){e1=num;count1++;}
            else if(count2==0){e2=num;count2++;}
            else{count1--;count2--;}
        }
        count1=0;count2=0;
        for(int i=0;i<nums.length;i++){
            int num=nums[i];
            if(num==e1) count1++;
            else if(num==e2) count2++;
        }  
        if(count1>nums.length/3) l.add(e1);
        if(count2>nums.length/3) l.add(e2);
        return l;

    }
}

File: 0229 majority-element-ii/solution.cs
public class Solution {
    public IList<int> MajorityElement(int[] nums) {
        IList<int> ans = new List<int>();
        int c1=0,c2=0,ct1=0,ct2=0;
        foreach(int i in nums) {
            if(c1==i) ct1++;
            else if(c2==i) ct2++;
            else if(ct1==0) {
                c1=i;
                ct1=1;
            } else if(ct2==0) {
                c2=i;
                ct2=1;
            } else {
                ct1--;
                ct2--;
            }
        }
        ct1=0;
        ct2=0;
        foreach(int i in nums) {
            if(c1==i) ct1++;
            else if(c2==i) ct2++;
        }
        int t=nums.Length/3;
        if(ct1>t) ans.Add(c1);
        if(ct2>t) ans.Add(c2);
        return ans;
    }
}
